private/buf/cmd/buf/command/alpha/plugin/pluginpush/pluginpush.go

```diff
@@ -216,44 +196,39 @@ func run(
 		retErr = multierr.Append(retErr, client.Close())
 	}()
 
-	var createdImage string
-	if len(flags.Image) > 0 {
-		tagResponse, err := client.Tag(ctx, flags.Image, pluginConfig)
+	var imageToTag string
+	if flags.Image != "" {
+		imageToTag = flags.Image
```

Comment:
The `--image` always takes precedence. If not specified, we fall back to looking for a pre-built image.tar in the directory where the `buf.plugin.yaml` is found.

Reply:
I'd error to avoid surprising behavior. But we don't need to get the UX ideal right now, let's see how this will be used.